src/Components/SmartComponents/SystemsExposedTable/SystemsExposedTable.js

```diff
@@ -114,7 +115,18 @@ const SystemsExposedTable = (props) => {
         DownloadReport.exec(fetchAffectedSystemsByCVE, params, format, 'systems-exposed');
     };
 
-    const showStatusModal = (cves, inventories) => {
+    const showStatusModal = (cves, inventoryIds) => {
+        // merge inventory ids and display_names into one object
+        let inventories = [];
```

Comment:
I believe you can avoid the two loops here. 

Every time we render the table we call [this](https://github.com/RedHatInsights/vulnerability-ui/blob/master/src/Helpers/CVEHelper.js#L18) function to generate the data for the table. Basically, we take the API response we add a few additional information and based on that the table is rendered.

The API response contains the attribute `display_name`, so I expect that each table row will "hold" that information because we ran the function above.

Also, because we perform the action (display modal) for a single row will be able to grab that information `onClick` 
Here is the  [onClick](https://github.com/RedHatInsights/vulnerability-ui/blob/master/src/Helpers/CVEHelper.js#L165)  function we can grab the `display_name`